### ISSUE TITLE
spline-reco lies about gradients (Trac #1701)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1693]
I3SplineRecoLikelihood advertises having gradients: 
http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/spline-reco/trunk/private/spline-reco/I3SplineRecoLikelihood.h#L85
but appears not to implement them:
http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/spline-reco/trunk/private/spline-reco/I3SplineRecoLikelihood.cxx#L285
(note the lack of any stores to `*gradient`).

Obviously it is not hard to change `HasGradient`'s result, but since it should be possible for gradients to be computed, I wanted there to at least be a record of this issue: having the feature complete and working would be even better than having it correctly marked as non-existent. 

Migrated from https://code.icecube.wisc.edu/ticket/1701
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:10", 
    "description": "I3SplineRecoLikelihood advertises having gradients: \nhttp://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/spline-reco/trunk/private/spline-reco/I3SplineRecoLikelihood.h#L85\nbut appears not to implement them:\nhttp://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/spline-reco/trunk/private/spline-reco/I3SplineRecoLikelihood.cxx#L285\n(note the lack of any stores to `*gradient`).\n\nObviously it is not hard to change `HasGradient`'s result, but since it should be possible for gradients to be computed, I wanted there to at least be a record of this issue: having the feature complete and working would be even better than having it correctly marked as non-existent. ", 
    "reporter": "cweaver", 
    "cc": "richman", 
    "resolution": "fixed", 
    "_ts": "1550067190995086", 
    "component": "combo reconstruction", 
    "summary": "spline-reco lies about gradients", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-05-12T20:41:08", 
    "milestone": "", 
    "owner": "richman", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160517T02:40:31: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "mvoge"

**Comment by icecube-trac** [comment]
Trac update at 20160517T02:41:54: olivas changed milestone from "" to "July 2016"

**Comment by icecube-trac** [comment]
Trac update at 20160816T20:02:56: 
* olivas changed cc from "" to "richman"
* olivas changed owner from "mvoge" to "kjmeagher"
* olivas commented:

> Voge is no longer with IceCube, so this ticket goes to the L3s...sorry KJ and Richman.

**Comment by icecube-trac** [comment]
Trac update at 20160817T07:16:24: 
* kjmeagher changed status from "assigned" to "closed"
* kjmeagher commented:

> In changeset 149292:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149292"
> Get spline-reco to tell the truth about its use of gradients. fixes #1693
> ```


* kjmeagher changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160817T17:29:28: richman commented:

> Would it be cleaner to remove the `HasGradient()` reimplementation, so it just falls back on the default (which returns `false`)?
> http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/gulliver/trunk/public/gulliver/I3EventLogLikelihoodBase.h#L127

**Comment by icecube-trac** [comment]
Trac update at 20160818T09:20:26: 
* kjmeagher changed resolution from "fixed" to ""
* kjmeagher changed status from "closed" to "reopened"

**Comment by icecube-trac** [comment]
Trac update at 20160818T09:21:58: 
* kjmeagher changed status from "reopened" to "assigned"
* kjmeagher commented:

> Yes, go a head and delete it 


* kjmeagher changed owner from "kjmeagher" to "richman"

**Comment by icecube-trac** [comment]
Trac update at 20160819T15:43:26: 
* richman changed resolution from "" to "fixed"
* richman changed status from "assigned" to "closed"
* richman commented:

> In changeset 149419:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149419"
> remove HasGradient(); fixes #1693
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:10: 
* olivas commented:

> Milestone July 2016 deleted


* olivas changed milestone from "July 2016" to ""
